### PR TITLE
65 missing readme check

### PR DESCRIPTION
**Pull request by eriktks**
Refs #65 

## Describe the changes made in this pull request

1. Added extra checks to react to non-standard cases for repository analysis
2. included a test file to automate checking handling of these cases 

## Instructions to review the pull request

1. Rebuild howfairis
2. Run `howfairis  https://gitlab.com/a.vijaykumar/bicgstab_l` to test the url from the issue. Remarks: the first error message of the issue was correct (the files in the repository are inaccessible, see: https://gitlab.com/api/v4/projects/a.vijaykumar%2Fbicgstab_l/repository/tree : 403 Forbidden); the second error message was solved by this PR
3. Run `pytest tests/test_cli.py` to  verify that five predefined cases are handled correctly
4. (optionally) run `howfairis url` on some other urls to check if they are handled correctly

### CODE REVIEW

**Comment by eriktks** [comment]
A note on the file livetests/test_cli.py: these tests were first put in the directory tests but the test for the correct url kept failing when run on github while it was fine when run on my machine. I moved the test file to the directory livetests to keep it from being run automatically and blocking the process. But perhaps there is a better solution?

**Comment by sverhoeven** [comment]
Please don't write unit tests (`/tests/*`) that fetch stuff from the Internet. These are slow and can fail unexpectedly or intermittently. Use `/clitests` or `/livetest` for those tests. Or use mocked requests in the unit tests.

**Comment by eriktks** [comment]
Yes, I already guessed that fetching the data was causing the problem. So it seems that the `/livetests` directory is the right place for `test.cli.py` after all.

File: howfairis/Repo.py [hunk 1]
```diff
@@ -25,13 +26,17 @@ def _check_assertions(url):
         assert url.startswith("https://"), "url should start with https://"
         assert True in [url.startswith("https://github.com"),
                         url.startswith("https://gitlab.com")], "Repository should be on github.com or on gitlab.com."
+        assert url.startswith("https://"), "url should start with https://"
+        assert True in [url.startswith("https://github.com"),
+                        url.startswith("https://gitlab.com")], "Repository should be on github.com or on " + \
+                                                               "gitlab.com"
```
Comment (sverhoeven) [inline howfairis/Repo.py:32]:
```suggestion
                                                               "gitlab.com."
```

File: livetests/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+def test_invalid_url():
+    runner = CliRunner()
+    result = runner.invoke(cli.cli, ["howfairis"])
+    expected_exception = "url should start with https://"
+    assert str(result.exception) == expected_exception
+
+def test_no_github():
+    runner = CliRunner()
+    result = runner.invoke(cli.cli, ["https://www.esciencecenter.nl"])
+    expected_exception = "Repository should be on github.com or on gitlab.com"
```
Comment (sverhoeven) [inline livetests/test_cli.py:21]:
```suggestion
    expected_exception = "Repository should be on github.com or on gitlab.com."
```

File: howfairis/Repo.py (hunk 1 above)
Comment (sverhoeven) [inline howfairis/Repo.py]:
```suggestion
```

File: howfairis/Repo.py [hunk 3]
```diff
@@ -25,13 +26,17 @@ def _check_assertions(url):
         assert url.startswith("https://"), "url should start with https://"
         assert True in [url.startswith("https://github.com"),
                         url.startswith("https://gitlab.com")], "Repository should be on github.com or on gitlab.com."
+        assert url.startswith("https://"), "url should start with https://"
+        assert True in [url.startswith("https://github.com"),
+                        url.startswith("https://gitlab.com")], "Repository should be on github.com or on " + \
+                                                               "gitlab.com."
```
Comment (sverhoeven) [inline howfairis/Repo.py:32]: Duplicate
```suggestion
```

**Review by sverhoeven — APPROVED** [review]
Test works for me. However example gitlab url changed its contents, but check is still working better then before.

**Comment by eriktks** [comment]
Thanks for the review and the updates!

**Pull request merged**